pkg/controller/machine/controller.go

```diff
@@ -145,6 +162,51 @@ func (r *ReconcileMachine) Reconcile(request reconcile.Request) (reconcile.Resul
 			return reconcile.Result{}, nil
 		}
 		klog.Infof("reconciling machine object %v triggers delete.", name)
+
+		// Drain node before deletion
+		// If a machine is not linked to a node, just delete the machine. Since a node
+		// can be unlinked from a machine when the node goes NotReady and is removed
+		// by cloud controller manager. In that case some machines would never get
+		// deleted without a manual intervention.
+		if _, exists := m.ObjectMeta.Annotations[ExcludeNodeDrainingAnnotation]; !exists && m.Status.NodeRef != nil {
+			if err := func() error {
```

Comment:
Why does this need to be a func?

Reply:
To avoid repeated `return reconcile.Result{}, err` every time an error is returned.

Reply:
Have to agree that this reads pretty weird. Can we just move it to an actual named function?

Reply:
Done